### PR TITLE
feat(cms): update gias import

### PR DESCRIPTION
**Pull request by serenaabbott11**
## Changes in this PR
- Added closed date, reason establishment opened, reason establishment closed, archived, archived at to support organisations
- Added closed date, opened date, archived, archived at to support establishment groups
- Added archived, archived at to local authorities

### CODE REVIEW

File: lib/school/information.rb [hunk 1]
```diff
@@ -62,6 +62,23 @@ def call
         exporter.call dataset
       end
 
+      urns_in_import = output.flatten.pluck(:urn)
+      Support::Organisation.where(urn: urns_in_import).update_all(archived: false, archived_at: nil)
+      Support::Organisation.where.not(urn: urns_in_import).where(archived_at: nil).update_all(archived: true, archived_at: Time.zone.now)
```
Comment (DevMagnataur) [inline lib/school/information.rb:67]: can we do this where("urn NOT IN (?) and archived_at=?", urns_in_import, nil)

File: lib/school/record_keeper.rb [hunk 2]
```diff
@@ -77,10 +80,10 @@ def type(record)
     # @param opened_date [String] DD-MM-YYYYY
     #
     # @return [Time, nil]
-    def parse_opened_date(opened_date)
-      return nil unless /\d{1,2}-\d{1,2}-\d{4}/.match? opened_date
+    def parse_date(opened_or_closed_date)
+      return nil unless /\d{1,2}-\d{1,2}-\d{4}/.match? opened_or_closed_date
 
-      Time.zone.parse(opened_date)
+      Time.zone.parse(opened_or_closed_date)
```
Comment (DevMagnataur) [inline lib/school/record_keeper.rb:86]: very well

File: lib/support/establishment_groups/information.rb [hunk 3]
```diff
@@ -63,6 +63,10 @@ def call
           exporter.call dataset
         end
 
+        uids_in_import = output.flatten.pluck(:uid)
+        Support::EstablishmentGroup.where(uid: uids_in_import).update_all(archived: false, archived_at: nil)
+        Support::EstablishmentGroup.where.not(uid: uids_in_import).where(archived_at: nil).update_all(archived: true, archived_at: Time.zone.now)
```
Comment (DevMagnataur) [inline lib/support/establishment_groups/information.rb:68]: same as above comment but only if you think its good then

File: lib/school/information.rb (hunk 1 above)
Comment (serenaabbott11) [reply lib/school/information.rb:67]: ooh thanks, will edit now and check it works

**Review by DevMagnataur — APPROVED** [review]

**Pull request merged**